src/complex/Utilities/DataArrayUtilities.hpp

```diff
@@ -904,4 +906,215 @@ class CopyTupleUsingIndexList
   IDataArray& m_NewCellArray;
   nonstd::span<const int64> m_NewIndices;
 };
+
+/**
+ * @brief This class is meant to make copying data from one IArray into another easier for the developer.
+ *
+ * An example use of these functions would be the following (where newCellData is an AttributeMatrix in dataStructure ):
+ *   ParallelTaskAlgorithm taskRunner;
+ *   for (const auto& [dataId, dataObject] : *newCellData)
+ *   {
+ *     auto* inputDataArray = dataStructure.getDataAs<IArray>(inputCellDataPath.createChildPath(name));
+ *     auto* destDataArray = dataStructure.getDataAs<IArray>(destCellDataPath.createChildPath(name));
+ *     auto* newDataArray = dataStructure.getDataAs<IArray>(newCellDataPath.createChildPath(name));
+ *     const IArray::ArrayType arrayType = destDataArray->getArrayType();
+ *     CopyFromArray::RunParallel<CopyFromArray::Combine>(arrayType, destDataArray, taskRunner, inputDataArray, newDataArray);
+ *   }
+ *   taskRunner.wait();
+ * @code
+ *
+ * @endcode
+ */
+class CopyFromArray
```

Comment:
@JDuffeyBQ I was trying to hide some of the functions so that the user has to go through the proper checks in order to do the actual copying (specifically with the AppendData function below).  Basically, in the case that we want to save the appended to a new geometry, you use the create actions in preflight to create arrays/neighborlists/string arrays for each valid cell data array.  The CreateNeighborListAction doesn't allocate so you would get down to that AppendData function and the `[]` operator for NeighborList tries to dereference a nullptr.

Reply:
Sorry I should have been more specific. I was thinking that it could be a namespace instead since it doesn't store any data.

Reply:
@JDuffeyBQ Yeah I get why we'd want to do a namespace instead. I was mostly just thinking that it makes it more clear to the user how the functions should be used together. I though it might be easy for a user of the code to see the AppendData function signature and try to use that without proper bounds checking and all the other steps to set up that function but maybe that's okay.  I've got a big comment at the top of this section explaining how it's meant to be used and if I comment the AppendData function well enough too then we can just put that on the caller to make sure they are using it properly.